shit-list.lic

```diff
@@ -0,0 +1,124 @@
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#shit-list
+=end
+
+custom_require.call(%w[drinfomon])
+
+class ShitListRecorder
+
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'add', regex: /add/i, description: 'Add name to list.' },
+        { name: 'name', regex: /\w+/i, description: 'Name to add.' },
+        { name: 'desc', regex: /.*$/i, description: 'Short description of why they were added in quotes.' }
+      ],
+      [
+        { name: 'find', regex: /find/i, description: 'Find if name exists and gives their description.' },
+        { name: 'name', regex: /\w+/i, description: 'Name of character to find.' }
+      ],
+      [
+        { name: 'rem', regex: /rem.*/i, description: 'Remove name from list.' },
+        { name: 'name', regex: /\w+/i, description: 'Name of character to remove.' }
+      ],
+      [
+        { name: 'clear', regex: /clear/i, description: 'Clear list! WARNING!' },
+        { name: 'confirm', regex: /confirm/i, description: 'CONFIRM YOUR CHOICE!.' }       
+      ],
+      []
+    ]
+
+    args = parse_args(arg_definitions)
+    @file_name = "#{checkname}_shit.list"
+
+    if args.add
+      add_new_shitter(args.name.downcase, args.desc)
+    elsif args.find
+      find_shitter_in_list(args.name.downcase)
+    elsif args.rem
+      remove_shitter(args.name.downcase)
+    elsif args.clear
+      clear_list(args.confirm)
```

Comment:
💡 Why pass in `args.confirm`? The arg parsing should ensure it's always there (since it is not marked as optional).

Reply:
I know, it was something I did early and just didn't remove, lol.